### PR TITLE
  Fixing issue 175: Misleading error message in the output window

### PR DESCRIPTION
**Pull request by raquelsa**
#175

### CODE REVIEW

File: src/LibraryManager.Vsix/Contracts/Logger.cs [hunk 1]
```diff
@@ -170,54 +170,60 @@ internal static void LogEventsSummary(IEnumerable<ILibraryOperationResult> total
...
 
-            LogEvent(messageText, LogLevel.Operation);
+                LogEvent(messageText, LogLevel.Operation);
+            }
         }
 
         private static string GetLibraryId(IEnumerable<ILibraryOperationResult> totalResults, OperationType operation)
         {
-            if (operation == OperationType.Uninstall || operation == OperationType.Upgrade)
+            if (totalResults != null && totalResults.Any())
```
Comment (ToddGrun) [inline src/LibraryManager.Vsix/Contracts/Logger.cs]: >if (totalResults != null && totalResults.Any()) [](start = 12, length = 47)

This isn't necessary, is it?

File: src/LibraryManager.Vsix/Shared/LibraryCommandService.cs [hunk 2]
```diff
@@ -67,7 +67,11 @@ public bool IsOperationInProgress
         public async Task RestoreAsync(string configFilePath, CancellationToken cancellationToken)
         {
             Dictionary<string, Manifest> manifests = await GetManifestFromConfigAsync(new[] { configFilePath }, cancellationToken);
-            await RestoreAsync(manifests, cancellationToken);
+
+            if (manifests.Any())
```
Comment (ToddGrun) [inline src/LibraryManager.Vsix/Shared/LibraryCommandService.cs]: >manifests.Any() [](start = 16, length = 15)

No need for linq here, Dictionary has a .Count property

**Comment by ToddGrun** [comment]
>                 Logger.LogEvent(LibraryManager.Resources.Text.Restore_OperationFailed + Environment.NewLine + ex.Message, LogLevel.Operation);

Is it ok to returned a partially filled manifest set?

---
Refers to: src/LibraryManager.Vsix/Shared/LibraryCommandService.cs:155 in d1df6ba. [](commit_id = d1df6ba36abfed142392ae526349ed91ec8a9703, deletion_comment = False)

**Comment by raquelsa** [comment]
>                 Logger.LogEvent(LibraryManager.Resources.Text.Restore_OperationFailed + Environment.NewLine + ex.Message, LogLevel.Operation);

That's a good point. With this implementation, we will restore the valid manifests and write an error to the output window for the ones that fails to parse (manifest == null)

---
In reply to: [400465509](https://github.com/aspnet/LibraryManager/pull/177#issuecomment-400465509) [](ancestors = 400465509)

---
Refers to: src/LibraryManager.Vsix/Shared/LibraryCommandService.cs:155 in d1df6ba. [](commit_id = d1df6ba36abfed142392ae526349ed91ec8a9703, deletion_comment = False)

**Comment by raquelsa** [comment]
>                 Logger.LogEvent(LibraryManager.Resources.Text.Restore_OperationFailed + Environment.NewLine + ex.Message, LogLevel.Operation);

I think it it is ok. I will log a bug to add more details on the error message so we know which manifest failed to parse. 

---
In reply to: [400485411](https://github.com/aspnet/LibraryManager/pull/177#issuecomment-400485411) [](ancestors = 400485411,400465509)

---
Refers to: src/LibraryManager.Vsix/Shared/LibraryCommandService.cs:155 in d1df6ba. [](commit_id = d1df6ba36abfed142392ae526349ed91ec8a9703, deletion_comment = False)

**Comment by ToddGrun** [comment]
>                 Logger.LogEvent(LibraryManager.Resources.Text.Restore_OperationFailed + Environment.NewLine + ex.Message, LogLevel.Operation);

One more question, we don't restore all valid manifests, just the valid ones we encounter before the failing one. Is that ok?

---
In reply to: [400485617](https://github.com/aspnet/LibraryManager/pull/177#issuecomment-400485617) [](ancestors = 400485617,400485411,400465509)

---
Refers to: src/LibraryManager.Vsix/Shared/LibraryCommandService.cs:155 in d1df6ba. [](commit_id = d1df6ba36abfed142392ae526349ed91ec8a9703, deletion_comment = False)

**Comment by raquelsa** [comment]
>                 Logger.LogEvent(LibraryManager.Resources.Text.Restore_OperationFailed + Environment.NewLine + ex.Message, LogLevel.Operation);

I will return null after logging.

---
In reply to: [400486135](https://github.com/aspnet/LibraryManager/pull/177#issuecomment-400486135) [](ancestors = 400486135,400485617,400485411,400465509)

---
Refers to: src/LibraryManager.Vsix/Shared/LibraryCommandService.cs:155 in d1df6ba. [](commit_id = d1df6ba36abfed142392ae526349ed91ec8a9703, deletion_comment = False)

**Pull request merged**